### DISCUSSION TITLE
Event player OR client disconnect dont works

### DISCUSSION THREAD

**Discussion by ghost** [saul/demofile#239] [Questions] [answered]
Hello. I found a bug.

var fs = require('fs');

var demo = require('demofile');

fs.readFile('demo.dem', function (err, buffer) {
	const demoFile = new demo.DemoFile();

	demoFile.gameEvents.on('player_spawn', e => {
		console.log(1); //WORKS
	});
	
	demoFile.gameEvents.on('client_disconnect', e => {
		console.log(2); //DONT WORKS
	});
	
	demoFile.gameEvents.on('player_disconnect', e => {
		console.log(3); //DONT WORKS
	});

	demoFile.parse(buffer);
});

**Comment by EnriqCG** [comment]
I am not entirely sure whether `client_disconnect` is a GameEvent or not.

**Comment by saul** [comment]
`client_disconnect` is a client-only event, so you won’t see it in demos. `player_disconnect` you should see, though. Do you have a demo that demonstrates this that you could upload?

**Comment by ghost** [comment]
Hello. yes, i have. [Demo](http://pglarena.ro/tools/download.php?file=csgotv/auto-20180607-232011-de_inferno.dem)
"player_disconnect" event never reaches it

**Comment by saul** [comment, marked as answer]
So it looks like player_disconnect is not networked - you'll have to listen for the player entity being deleted. This is guaranteed to only happen at disconnect time:

```js
      demoFile.entities.on('beforeremove', e => {
        if (e.entity.serverClass.name !== 'CCSPlayer') {
          return;
        }

        console.log('%s left the game', e.entity.name);
      });
```
